package.json

```diff
@@ -62,8 +62,8 @@
   },
   "dependencies": {
     "@types/jwt-decode": "^2.2.1",
-    "balena-errors": "^4.2.1",
-    "balena-settings-storage": "^6.0.0",
+    "balena-errors": "^4.7.1",
+    "balena-settings-storage": "^7.0.0",
```

Comment:
I see that `getStorage` is called on this line:
https://github.com/balena-io-modules/balena-auth/blob/v4.0.2/lib/auth.ts#L40
```js
	constructor({ dataDirectory, tokenKey = 'token' }: BalenaAuthOptions = {}) {
		this.storage = getStorage({ dataDirectory });
		this.tokenKey = tokenKey;
	}
```
`balena-settings-storage` v7 raises an error where previously it would fail silently. What is the consequence of an uncaught error inside a constructor, as above? Does the error need to be caught? Consider not only the balena CLI, but all possible users of balena-auth, including the balena SDK and users of the balena SDK (balena users' own applications that directly use the SDK).

Also: the commit for this PR says the change type is minor. What's the reasoning? I usually understand `minor` as a new feature (users can do something useful that they couldn't before), major as breaking backwards compatibility, and patch if a bug fix or "harmless" (not impactful) dependency updates.

Reply:
`getStorage` does not raises an exception, the exception only happens only when trying to get a value from a file without the expected permission. https://github.com/balena-io-modules/balena-settings-storage/blob/master/lib/storage.ts#L101

In this particular case, only on https://github.com/balena-io-modules/balena-auth/blob/master/lib/auth.ts#L212, which already can throw an `BalenaMalformedToken` error.

I understand your point. I only tested it on `balena-cli` and I didn't take into considaration any other scenario. But, as the affected methods are already throwing custom errors, I don't this this is a breaking change.

Reply:
@thgreasi, any concerns, thinking of the SDK?

Reply:
@josecoelho @pdcastro let's open a draft SDK PR using the pre-release version of this PR and see whether the tests pass.

Reply:
> using the pre-release version of this PR 

@josecoelho, in case you wonder (it was news to me when I joined Balena!), :-) the "pre-release version" of this PR can be listed in the Versions tab of the npm registry page: https://www.npmjs.com/package/balena-auth

So it would mean opening a draft `balena-io/balena-sdk` PR and changing the following line in its `package.json`:
https://github.com/balena-io/balena-sdk/blob/v15.17.0/package.json#L102

From: `"balena-auth": "^4.0.2",`
To: `"balena-auth": "^4.1.0-update-balena-settings-75ee6cdf7f9445792d8339eb7c4b9df4e0304cc2",`

And seeing whether the SDK's tests pass (rather than `balena-auth`'s tests).

Reply:
https://github.com/balena-io/balena-sdk/pull/1018 created.

We will probably have to update it anyway. I was testing locally using `npm link` which can be tricky when dealing with those dependencies of dependencies. If we don't upgrade `balena-sdk` it will keep using an old version of `balena-auth` with an old version of `balena-settings-storage`.

You can check the flow here: https://github.com/balena-io/balena-cli/issues/1667#issuecomment-700358631

Reply:
We might have to upgrade `balena-request` as well.
https://github.com/balena-io-modules/balena-request/blob/master/package.json

Reply:
Just in case, https://github.com/balena-io-modules/balena-request/pull/152 created as well.

Reply:
> If we don't upgrade balena-sdk it will keep using an old version of balena-auth with an old version of balena-settings-storage.

@joehounsham that's only try in the CLI b/c is uses a shrinkwrap though.
1. In the mean time we would want to know whether the newer balena-auth breaks the sdk's CI tests, which will be picking up the new version once this PR gets merged
2. Even though it's technically possible to bump sub-dependencies in the cli w/o bumping the sdk, it's not the best approach. Bumping the SDK is more explicit and safer as well, since we can be sure that the code in the new version has been tested when using the latest version of each sub-dependency.

Reply:
Getting there. I believe this one is ready to go.

Reply:
@thgreasi @pdcastro

And then? https://github.com/balena-io-modules/balena-request/pull/152
And then?  https://github.com/balena-io/balena-sdk/pull/1018
And then? https://github.com/balena-io/balena-cli/pull/2050
And then?!

![chinesefood](https://user-images.githubusercontent.com/1987013/98854547-91e52880-24bf-11eb-8537-12cae3fd8caa.gif)

Reply:
That's correct @josecoelho